pkg/syncer/resourcesync/controller.go

```diff
@@ -152,7 +118,75 @@ func NewController(
 	return c, nil
 }
 
-func (c *Controller) enqueueSyncTarget(obj interface{}, logger logr.Logger) {
+var _ informer.GVRSource = (*controller)(nil)
+
+// controller is a control loop that watches synctarget. It starts/stops spec syncer and status syncer
+// per gvr based on synctarget.Status.SyncedResources.
+// All the spec/status syncer share the same downstreamNSInformer and upstreamSecretInformer. Informers
+// for gvr is started separated for each syncer.
+type controller struct {
+	queue                workqueue.RateLimitingInterface
+	downstreamKubeClient kubernetes.Interface
+
+	cachedDiscovery discovery.CachedDiscoveryInterface
```

Comment:
discovery of what?

Reply:
Renamed to `upstreamSyncerDiscoveryClient`